ops/testing.py

```diff
@@ -1804,8 +1865,28 @@ def action_log(self, message):  # type:ignore
     def action_fail(self, message=''):  # type:ignore
         raise NotImplementedError(self.action_fail)  # type:ignore
 
-    def network_get(self, endpoint_name, relation_id=None):  # type:ignore
-        raise NotImplementedError(self.network_get)  # type:ignore
+    def network_get(self, binding_name: str, relation_id: Optional[int] = None) -> '_NetworkDict':
+        if relation_id is None:
+            data = self._networks.get((binding_name, None))
+            if data is not None:
+                # Get the default data if it exists
+                return data
+            # Otherwise get the data with the lowest relation_id as the default
+            # TODO: does this even make sense?
+            datas = sorted((r, data) for (b, r), data in self._networks.items()
```

Comment:
it's hard to see what you're doing here, it doesn't help that you're using single-character variables :/

Reply:
Fair. :-) This has gone away now that it's updated.